### DISCUSSION TITLE
Proper way of integration with Kratos for invite only system?

### DISCUSSION THREAD

**Discussion by trobinpl** [ory/kratos#1342] [Q&A] [answered]
I'm trying to get my head around on how to integrate Kratos as identity provider for my use case which is as web application where you can't register by yourself, but you have to be invited by the administrator. So far when user is invited I create an identity using Kratos Admin API which works fine. Then I want to send user an email asking him to confirm his account (and get some information about him in the process). I was thinking about using the recovery flow for that, but it didn't seem like appropriate solution. Then I was trying to implement the verify flow, but I already have an e-mail service in place which I would like to be responsible for sending all emails coming out of my application. During the verify flow, once user provides his e-mail the flow goes to `sent_email` state. Is it possible just to generate the link and make the sending on the application side? In general - how such use case should be implemented?

**Comment by Benehiko** [comment, marked as answer]
Hi @trobinpl,

Thanks for reaching out :) 

In the case of you creating the user account, you should the following:
1. Create account via Admin API
2. Create a recovery link via Admin API
3. Email recovery link to users

This will force the user to go through the recovery flow which will ask them to set a password.

Step 3 should be done inside your application manually via your email service.

You can checkout our docs for more information [here](https://www.ory.sh/kratos/docs/admin/managing-users-identities#invite-a-user)
